pkgs/development/python-modules/matrix-nio/default.nix

```diff
@@ -52,31 +54,34 @@ buildPythonPackage rec {
     hash = "sha256-XlswVHLvKOi1qr+I7Mbm4IBjn1DG7glgDsNY48NA5Ew=";
   };
 
-  nativeBuildInputs = [ poetry-core ];
-
-  propagatedBuildInputs = [
-    aiofiles
-    aiohttp
-    aiohttp-socks
-    h11
-    h2
-    jsonschema
-    pycryptodome
-    unpaddedbase64
+  patches = [
+    # Ignore olm import failures when testing
+    ./allow-tests-without-olm.patch
   ];
 
-  pythonRelaxDeps = [
-    "aiohttp-socks" # Pending matrix-nio/matrix-nio#516
-  ];
+  nativeBuildInputs = [ poetry-core ];
 
-  passthru.optional-dependencies = {
-    e2e = [
```

Comment:
Please keep the optional-dependencies. That's the right approach here https://github.com/matrix-nio/matrix-nio/blob/0.25.0/pyproject.toml#L37